modules/database/src/ioc/db/dbTest.c

```diff
@@ -305,8 +306,25 @@ long dbgrep(const char *pmask)
         status = dbFirstRecord(pdbentry);
         while (!status) {
             char *pname = dbGetRecordName(pdbentry);
-            if (epicsStrGlobMatch(pname, pmask))
-                puts(pname);
+            if (epicsStrGlobMatch(pname, pmask)) {
+                if (field == NULL) {
+                    puts(pname);
+                }
+                else {
+                    const char *pvalue = "<field not present>";
+                    status = dbFindField(pdbentry, field);
+                    if (status) {
+                        if (!strcmp(field, "recordType")) {
+                            pvalue = dbGetRecordTypeName(pdbentry);
+                        }
```

Comment:
I don't believe this is necessary, try requesting a field named `RTYP`.

Reply:
I took this from the `dbl` code that seems to allow both `RTYP` and `recordType`, is the `recordType`  just historical and pre-dates `RTYP` and so for any new code like this there is no need/expectation of `recordType` working?

Reply:
Oops, well I have no memory of that code snippet in `dbl`, despite working on it about 17 years ago. It almost certainly predates the addition of the `RTYP` record attribute, which was a more general solution to record-type identification that works remotely. The CA library also had a special request type for the record type-name of a channel, which nobody uses now. I guess this does no harm, but unless it's documented for `dbl` I'd probably consider removing it from there sometime.